packages/react-tabs/src/components/TabList/useTabList.ts

```diff
@@ -31,67 +31,33 @@ export const useTabList_unstable = (props: TabListProps, ref: React.Ref<HTMLElem
     initialState: undefined,
   });
 
+  // considered usePrevious, but it is sensitive to re-renders
+  // this could cause the previous to move to current in the case where the tab list re-renders.
+  const [previousSelectedValue, setPreviousSelectedValue] = useControllableState<TabValue | undefined>({
+    state: undefined,
+    defaultState: undefined,
+    initialState: undefined,
+  });
+
   const onSelect = useEventCallback((event: SelectTabEvent, data: SelectTabData) => {
-    onTabSelect?.(event, data);
+    setPreviousSelectedValue(selectedValue);
     setSelectedValue(data.value);
+    onTabSelect?.(event, data);
   });
 
-  // provide force recalculation of the selection indicator rectangle
-  const [calcRect, setCalcRect] = React.useState(0);
-  const recalcRect = useConst(() => () => setCalcRect(value => ++value));
-
-  // when this list or any tab resizes, recalculate the selection indicator rectangle
-  const resizeObserver = React.useMemo(
-    () =>
-      new ResizeObserver(entries => {
-        recalcRect();
-      }),
-    [recalcRect],
-  );
-
-  // observe this list for resize
-  React.useEffect(() => {
-    const currentRef = innerRef.current;
-    currentRef && resizeObserver.observe(currentRef);
-
-    return () => {
-      currentRef && resizeObserver.unobserve(currentRef);
-    };
-  }, [resizeObserver]);
-
   // when tabs register their refs, observe them for resize
   const registeredTabs = React.useRef<Record<string, RegisterTabData>>({});
 
   const onRegister = useEventCallback((data: RegisterTabData) => {
     registeredTabs.current[JSON.stringify(data.value)] = data;
-    data.ref?.current && resizeObserver.observe(data.ref.current);
-    recalcRect();
+    data.ref?.current;
```

Comment:
```suggestion
```

Honestly, it's unclear what does `data.ref?.current`. Is it required? Looks like cruft...

Reply:
Realized that I have removed the need for resize observer and was able to remove the mock of it in the test.